src/Util/Util.php

```diff
@@ -74,20 +74,19 @@ public static function getXhtmlFromContent( $lang, $content, $title = ' ' ) {
 		return $html . '><head><meta content="application/xhtml+xml;charset=UTF-8" http-equiv="default-style" /><link type="text/css" rel="stylesheet" href="main.css" /><title>' . $title . '</title></head><body>' . $content . '</body></html>';
 	}
 
-	public static function getTempFile( $lang, $name ) {
+	public static function getTempFile( Api $api, $lang, $name ) {
```

Comment:
will $lang ever be different than the language already set in the api instance? Looks to me that $lang was needed here 'cause the object was being constructed inside the method

Reply:
If you pass `beta` to Api, it sets the language as the user's default language but the temp files are saved under `beta` in the cache dir. I had the same question as @dayllanmaza but since I was testing using `beta`, I realized that this is a case where the Api lang is different than the $lang value.

Reply:
@samwilson follow up patch is better yeah (if we need to change anything)
@cimurah I'm not sure I understand, can you elaborate more?

Reply:
There's some values that can be passed as the `$lang` like: `www`, `wl`, `wikilivres`, and `beta`. For instance, when we use the `beta` value, the Api.php makes the book request to `en.wikisource.beta.wmflabs.org` then it sets the `$lang = ''` so the temp files are grabbed from the `cached` root directory.

After analyzing this a bit further, it might be fine to use the `$lang` in the Api instance, but we should do it in a follow up since the logic around the cache is a bit convoluted.

Reply:
oh I see, thank you for explaining. A follow up patch sounds good.